capi/geos_ts_c.cpp

```diff
@@ -3745,6 +3745,14 @@ extern "C" {
         });
     }
 
+    void
+    GEOSGeoJSONWriter_setOutputDimension_r(GEOSContextHandle_t extHandle, GeoJSONWriter* writer, int dim)
+    {
+        execute(extHandle, [&]() {
+            writer->setOutputDimension(static_cast<uint8_t>(dim));
```

Comment:
Seems like this will overflow pretty easily...

Reply:
I just copied the implementation from `GEOSWKTWriter_setOutputDimension_r` and `GEOSWKBWriter_setOutputDimension_r`, but yeah that's not exactly safe.

Reply:
The only dimensions supported are 2, 3 and 4.  And `uint8_t` is used pretty widely for dimensions in the codebase.

Reply:
Yeah, I guess if someone overflows it they just get a garbage number, which in turn is what they put in (garbage) so, it's all on them. As long as somewhere deeper in the code base we are checking for ==2 or ==3 or ==4

Reply:
There's a check in [`GeoJSONWriter::setOutputDimension`](https://github.com/libgeos/geos/blob/main/src/io/GeoJSONWriter.cpp#L49), so that seems pretty safe.